File: basicsProgramming/module1/practice1/pascal/task1-4.pas
program Task1_4;

var
    num: integer;
    isValid: boolean;
    i, sumNums, compNums, reverseNum, modValue: integer;

begin
    writeln('Введите трехзначное число');
    {$I-}
    readln(num);
    {$I+}

    if IOResult <> 0 then
    begin
        writeln('Ошибка: введите целое и трехзначное число');
        Halt(1);
    end;

    isValid := (num >= 100) and (num <= 999);

    if not isValid then
    begin
        writeln('Ошибка: число не трехзначное');
        Halt(1);
    end;

    sumNums:= 0;
    compNums := 1;
    reverseNum := 0;

    for i := 1 to 3 do
    begin
        modValue := num mod 10;

        sumNums := sumNums + modValue;
        compNums := compNums * modValue;
        reverseNum := reverseNum * 10 + modValue;

        num := num div 10;
    end;

    writeln('Сумма чисел: ', sumNums);
    writeln('Произведение чисел: ', compNums);
    writeln('Перевернутое число: ', reverseNum);
end.

File: basicsProgramming/module1/practice1/c/task1-4.c
#include <stdio.h>
#include <stdbool.h>

void clearBuffer(){
    int c;
    while ((c = getchar()) != '\n' && c != EOF);
}

void task1_4()
{
    int num, sum = 0, reverse = 0, composition = 1, modValue;
    bool isValid = false;

    while(!isValid){
        printf("Введите целое трехзначное число ");

        if(scanf("%d", &num) != 1){
            printf("Ошибка: введите целое трехзначное число \n");
            clearBuffer();
            continue;
        }

        isValid = (num >= 100 && num <= 999) || (num >= -999 && num <= -100);

        if(!isValid){
            printf("Ошибка: число не трехзначное \n");
            continue;
        }
    }

    for (int i = 0; i < 3; i++)
    {
        modValue = num % 10;

        sum += modValue;
        composition *= modValue;
        reverse = reverse * 10 + modValue;

        num = num / 10;
    }
    
    printf("Сумма чисел = %d\n", sum);
    printf("Произведение чисел = %d\n", composition);
    printf("Переворот числа = %d\n", reverse);
}

int main()
{
    task1_4();
    return 0;
}
